web/pdf_page_view.js

```diff
@@ -1036,19 +1036,22 @@ class PDFPageView {
     const sfx = approximateFraction(outputScale.sx);
     const sfy = approximateFraction(outputScale.sy);
 
-    canvas.width = floorToDivide(width * outputScale.sx, sfx[0]);
-    canvas.height = floorToDivide(height * outputScale.sy, sfy[0]);
-    const { style } = canvas;
-    style.width = floorToDivide(width, sfx[1]) + "px";
-    style.height = floorToDivide(height, sfy[1]) + "px";
+    const newWidth = (canvas.width = floorToDivide(
+      width * outputScale.sx,
+      sfx[0]
+    ));
+    const newHeight = (canvas.height = floorToDivide(
+      height * outputScale.sy,
+      sfy[0]
+    ));
+    const scaleX = newWidth / width;
+    const scaleY = newHeight / height;
 
     // Add the viewport so it's known what it was originally drawn with.
     this.#viewportMap.set(canvas, viewport);
 
     // Rendering area
-    const transform = outputScale.scaled
-      ? [outputScale.sx, 0, 0, outputScale.sy, 0, 0]
-      : null;
+    const transform = outputScale.scaled ? [scaleX, 0, 0, scaleY, 0, 0] : null;
```

Comment:
Given the comment above, this should be reverted.
```suggestion
    const transform = outputScale.scaled
      ? [outputScale.sx, 0, 0, outputScale.sy, 0, 0]
      : null;
```